src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodoElement } from '../TodoElement';
+
+enum FilterType {
+  All,
+  Active,
+  Completed,
+}
+
+type Props = {
+  todos: Todo[];
+  tempTodo: Todo | undefined;
+  onDelete: (id: number) => void;
+  loadingIds: number[];
+  onDeleteCompleted: () => void;
+  onUpdateTodo: (id: number, data: Partial<Todo>) => void;
+};
+
+const getTodos = (todos: Todo[], filterMethod: FilterType): Todo[] => {
+  switch (filterMethod) {
+    case FilterType.Active:
+      return todos.filter(todo => !todo.completed);
+
+    case FilterType.Completed:
+      return todos.filter(todo => todo.completed);
+
+    default:
+      return todos;
+  }
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  tempTodo,
+  onDelete,
+  loadingIds,
+  onDeleteCompleted,
+  onUpdateTodo,
+}) => {
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+
+  const visibleTodos = getTodos(todos, filterType);
+
+  const remainingTodos = todos.filter(todo => !todo.completed).length;
+
+  const completedTodos = todos.filter(todo => todo.completed).length;
+
+  return (
+    <>
+      <section className="todoapp__main">
+        {visibleTodos.map(todo => {
+          const isLoading = loadingIds.some(id => id === todo.id);
+
+          return (
+            <TodoElement
+              todo={todo}
+              key={todo.id}
+              onDelete={onDelete}
+              isLoading={isLoading}
+              onUpdateTodo={onUpdateTodo}
+            />
+          );
+        })}
+      </section>
+
+      {tempTodo && (
+        <TodoElement
+          todo={tempTodo}
+          key={tempTodo.id}
+          onDelete={() => {}}
+          isLoading
+          onUpdateTodo={() => {}}
+        />
+      )}
+
+      <footer className="todoapp__footer">
+        <span className="todo-count">
+          {`${remainingTodos} ${remainingTodos === 1 ? 'item' : 'items'} left`}
+        </span>
+
+        {/* Active filter should have a 'selected' class */}
+        <nav className="filter">
+          <a
+            href="#/"
+            className={classNames(
+              'filter__link',
+              {
+                selected: filterType === FilterType.All,
+              },
+            )}
+            onClick={() => {
+              if (filterType !== FilterType.All) {
+                setFilterType(FilterType.All);
+              }
+            }}
+          >
+            All
+          </a>
```

Comment:
let's create component for it which will render needed link based on `FilterType` that was passed as a prop

Reply:
and you can map your enum using `Object.values`